### PR TITLE
Update of tf's assignment

### PR DESCRIPTION
**Pull request by pchauchat**
This PR adds a new use case to tf's `assignment` and the sum version (which was already accessible for numpy and pytorch): vectorization with non-constant values. This simply means that , if `ndim(x) == 3`:
`x = gs.assignment(x, [1, 2, 3], (0, 0), axis=2) <=> x[0, 0, :] = [1, 2, 3]`.

The docstring of assignment is also updated to try and cover its use cases.

### CODE REVIEW

File: geomstats/_backend/tensorflow/__init__.py [hunk 1]
```diff
@@ -325,7 +323,18 @@ def assignment_by_sum(x, values, indices, axis=0):
...
+    Most examples translate as
+    assignment(x, indices, values) <=> x[indices] = x[indices] + values
+    Some special cases are given by vectorisation.
+    (Beware that copies are always returned).
+    if ndim(x) == 3, assignment(x, 1, (1, 0), 1) <=> x[1, :, 0] += 1
+    if ndim(x) == 2, assignment(x, [1, 2], [(0, 1), (2, 3)]) <=>
+                        x[((0, 2), (1, 3))] += [1, 2]
     """
     if _is_boolean(indices):
         if ndim(array(indices)) > 1:
```
Comment (ninamiolane) [inline geomstats/_backend/tensorflow/__init__.py:340]: This is not tested.

File: geomstats/_backend/tensorflow/__init__.py [hunk 2]
```diff
@@ -344,7 +353,11 @@ def assignment_by_sum(x, values, indices, axis=0):
     if tf.is_tensor(indices):
         single_index = ndim(indices) <= 1 and sum(indices.shape) <= ndim(x)
```
Comment (ninamiolane) [inline geomstats/_backend/tensorflow/__init__.py:354]: This is not tested.

File: geomstats/_backend/tensorflow/__init__.py [hunk 3]
```diff
@@ -344,7 +353,11 @@ def assignment_by_sum(x, values, indices, axis=0):
...
     if single_index:
-        indices = [indices]
+        if len(values) > 1:
+            indices = [tuple(list(indices[:axis]) + [i] + list(indices[axis:]))
+                       for i in range(x.shape[axis])]
+        else:
+            indices = [indices]
```
Comment (ninamiolane) [inline geomstats/_backend/tensorflow/__init__.py:360]: This is not tested.

File: geomstats/_backend/tensorflow/__init__.py [hunk 4]
```diff
@@ -344,7 +353,11 @@ def assignment_by_sum(x, values, indices, axis=0):
...
 
     if len(values) != len(indices):
         raise ValueError('Either one value or as many values as indices')
```
Comment (ninamiolane) [inline geomstats/_backend/tensorflow/__init__.py:363]: This is not tested.

File: geomstats/_backend/tensorflow/__init__.py [hunk 5]
```diff
@@ -378,10 +391,6 @@ def _assignment_single_value(x, value, indices, axis=0):
         Copy of x where value was assigned at all indices (and possibly
         along an axis).
     """
-    if _is_boolean(indices):
-        indices = [indices]
-        indices = [index for index, val in enumerate(indices) if val]
-
     single_index = not isinstance(indices, list)
     if tf.is_tensor(indices):
         single_index = ndim(indices) <= 1 and sum(indices.shape) <= ndim(x)
```
Comment (ninamiolane) [inline geomstats/_backend/tensorflow/__init__.py:396]: This is not tested.

File: geomstats/_backend/tensorflow/__init__.py [hunk 6]
```diff
@@ -451,7 +473,11 @@ def assignment(x, values, indices, axis=0):
...
```
Comment (ninamiolane) [inline geomstats/_backend/tensorflow/__init__.py:474]: This is not tested.

File: geomstats/_backend/tensorflow/__init__.py [hunk 7]
```diff
@@ -451,7 +473,11 @@ def assignment(x, values, indices, axis=0):
...
     if single_index:
-        indices = [indices]
+        if len(values) > 1:
+            indices = [tuple(list(indices[:axis]) + [i] + list(indices[axis:]))
+                       for i in range(x.shape[axis])]
+        else:
+            indices = [indices]
```
Comment (ninamiolane) [inline geomstats/_backend/tensorflow/__init__.py:480]: This is not tested.

File: geomstats/_backend/tensorflow/__init__.py [hunk 8]
```diff
@@ -451,7 +473,11 @@ def assignment(x, values, indices, axis=0):
...
 
     if len(values) != len(indices):
         raise ValueError('Either one value or as many values as indices')
```
Comment (ninamiolane) [inline geomstats/_backend/tensorflow/__init__.py:483]: This is not tested.

**Comment by pchauchat** [comment]
Normally the number of untested lines decreased. I think most of the remaining ones are cases of `assignment_by_sum` which are already tested in `assignment`. The associated tests could be simply copied, but refactoring could also be interesting in this case, although this is not as trivial. For a subsequent PR ?
